TaskManagement/src/TaskManagment/functionality/TaskDBHelper.java

```
@@ -0,0 +1,474 @@
+            ps.setLong(7, task.getCreatedByUserId());
+            ps.setString(8, task.getCategory().toString());
+            ps.setString(9, task.getStatus().toString());
+            ps.executeUpdate();
+            ps.close();
+            con.close();
+            if (CollectionUtils.isNotEmpty(task.getPreRequisites())) {
+                ps = con.prepareStatement("select MAX(task_id) as last_id FROM task_management_system.task;");
+                ResultSet rset = ps.executeQuery();
+                ps.close();
+                con.close();
+
+                if (rset.next()) {
+                    addPreRequisiteTask(rset.getLong("last_id"), task.getPreRequisites());
+                }
+            }
+
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static ArrayList<Task> showTasks(long userId) {
+        ArrayList<Task> tasks = new ArrayList<>();
+        Connection con = Connections.getConnection();
+        try {
+            PreparedStatement ps = con.prepareStatement("SELECT * FROM task_management_system.task where assigned_user_id=?");
+            ps.setLong(1, userId);
+            ResultSet rs = ps.executeQuery();
+            while (rs.next()) {
+                ArrayList<PreRequisite> pre_tasks = getListOfPreTask(rs.getLong("task_id"));
+                Task task = new Task(rs.getLong("task_id"), rs.getString("task_name"), rs.getString("description"),
+                        rs.getString("tag"), rs.getDate("due_date").toLocalDate(), rs.getInt("priority_level"),
+                        pre_tasks, rs.getLong("assigned_user_id"), rs.getLong("created_by_user_id"), rs.getString("category"),
+                        rs.getString("status"));
+                tasks.add(task);
+            }
+            rs.close();
+            ps.close();
+            con.close();
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
```

We need to handle exceptions in a better way, it should return an error code and use finally work to close the opened connections